<a href="https://colab.research.google.com/github/19PA1A0466/COVID-19/blob/master/AB_vs_DT_vs_RF.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [0]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from sklearn.model_selection import train_test_split
from sklearn.metrics import confusion_matrix
from sklearn.preprocessing import LabelEncoder
from sklearn.ensemble import AdaBoostClassifier
from sklearn.tree import DecisionTreeClassifier

In [0]:
patients=pd.read_csv('diabetes.csv')

In [3]:

patients

Pregnancies  Glucose  ...  Age  Outcome
0              6      148  ...   50        1
1              1       85  ...   31        0
2              8      183  ...   32        1
3              1       89  ...   21        0
4              0      137  ...   33        1
..           ...      ...  ...  ...      ...
763           10      101  ...   63        0
764            2      122  ...   27        0
765            5      121  ...   30        0
766            1      126  ...   47        1
767            1       93  ...   23        0

[768 rows x 9 columns]

In [4]:
print(patients.columns)

Index(['Pregnancies', 'Glucose', 'BloodPressure', 'SkinThickness', 'Insulin',
       'BMI', 'DiabetesPedigreeFunction', 'Age', 'Outcome'],
      dtype='object')


In [5]:
patients.head()

Pregnancies  Glucose  BloodPressure  ...  DiabetesPedigreeFunction  Age  Outcome
0            6      148             72  ...                     0.627   50        1
1            1       85             66  ...                     0.351   31        0
2            8      183             64  ...                     0.672   32        1
3            1       89             66  ...                     0.167   21        0
4            0      137             40  ...                     2.288   33        1

[5 rows x 9 columns]

In [6]:



print(patients.groupby("Outcome").size())

Outcome
0    500
1    268
dtype: int64


In [7]:

patients.describe()

Pregnancies     Glucose  ...         Age     Outcome
count   768.000000  768.000000  ...  768.000000  768.000000
mean      3.845052  120.894531  ...   33.240885    0.348958
std       3.369578   31.972618  ...   11.760232    0.476951
min       0.000000    0.000000  ...   21.000000    0.000000
25%       1.000000   99.000000  ...   24.000000    0.000000
50%       3.000000  117.000000  ...   29.000000    0.000000
75%       6.000000  140.250000  ...   41.000000    1.000000
max      17.000000  199.000000  ...   81.000000    1.000000

[8 rows x 9 columns]

Adaboost Classifier

In [0]:
#divide the data for training and test
X_train, X_test, y_train, y_test = train_test_split(patients.loc[:, patients.columns != 'Outcome'], patients['Outcome'], stratify=patients['Outcome'], random_state=66)

In [9]:
from sklearn.ensemble import AdaBoostClassifier
ad=AdaBoostClassifier(random_state=0,n_estimators=150,base_estimator= None,learning_rate=1)
ad.fit(X_train,y_train)
print("Accuracy on training set: {:.2f}".format(ad.score(X_train, y_train)))
print("Accuracy on test set: {:.2f}".format(ad.score(X_test, y_test)))

Accuracy on training set: 0.86
Accuracy on test set: 0.76


In [28]:
ad=AdaBoostClassifier(DecisionTreeClassifier(max_depth=1),n_estimators=150)
ad.fit(X_train,y_train)

AdaBoostClassifier(algorithm='SAMME.R',
                   base_estimator=DecisionTreeClassifier(ccp_alpha=0.0,
                                                         class_weight=None,
                                                         criterion='gini',
                                                         max_depth=1,
                                                         max_features=None,
                                                         max_leaf_nodes=None,
                                                         min_impurity_decrease=0.0,
                                                         min_impurity_split=None,
                                                         min_samples_leaf=1,
                                                         min_samples_split=2,
                                                         min_weight_fraction_leaf=0.0,
                                                         presort='deprecated',
                          

In [0]:

#predictions of test data
predictions = ad.predict(X_test)

In [30]:
from sklearn import metrics
Accuracy=metrics.accuracy_score(y_test,predictions)
Accuracy

0.7552083333333334

Here accuracy is 75%

Decision Tree Classifier

In [0]:
from sklearn.tree import DecisionTreeClassifier

In [32]:
tree = DecisionTreeClassifier(random_state=0)
tree.fit(X_train, y_train)
print("Accuracy on training set: {:.3f}".format(tree.score(X_train, y_train)))
print("Accuracy on test set: {:.3f}".format(tree.score(X_test, y_test)))

Accuracy on training set: 1.000
Accuracy on test set: 0.714


In [33]:
tree = DecisionTreeClassifier(max_depth=3, random_state=0)
tree.fit(X_train, y_train)
print("Accuracy on training set: {:.3f}".format(tree.score(X_train, y_train)))
print("Accuracy on test set: {:.3f}".format(tree.score(X_test, y_test)))

Accuracy on training set: 0.773
Accuracy on test set: 0.740


Here accuarcy is 74%

Random Forest

In [0]:
from sklearn.ensemble import RandomForestClassifier

In [35]:

#applying random forest for 150 trees
rf = RandomForestClassifier(n_estimators=150, random_state=0)
rf.fit(X_train, y_train)
print("Accuracy on training set: {:.2f}".format(rf.score(X_train, y_train)))
print("Accuracy on test set: {:.2f}".format(rf.score(X_test, y_test)))

Accuracy on training set: 1.00
Accuracy on test set: 0.79


Here accuracy is 79%

By using above three techniques we can see that "random forest" got an accuracy of 79%.Where as using  "Adaboost classifier"  got an accuracy of 75%.Where as using "Decision tree" we got an accuracy of 74%

So,finally Random forest did well